### PR TITLE
wstunnel-rust: init at 9.2.2

### PR DESCRIPTION
**Pull request by Aleksanaa**
## Description of changes

The Rust rewrite of [wstunnel](https://search.nixos.org/packages?channel=unstable&show=wstunnel&from=0&size=50&sort=relevance&type=packages&query=wstunnel) from the same author. The  Haskell version may no longer be maintained. The command line arguments are not compatible though, so I dare not replace it directly.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by kirillrdy** [comment]
i wonder if we can hijack the name 

https://repology.org/project/wstunnel/versions 

cc @alyaeanyx

**Comment by alyaeanyx** [comment]
Not sure whether hijacking the name is a good idea. Judging from a quick look at the new readme of wstunnel, the command line options are partially compatible (for instance -L remains the same), so scripts using wstunnel might break in unexpected ways a while after the update. IMO the best approach to this situation would be to first merge this PR, and then remove the old package and rewire the wstunnel module to the new version in a separate PR.

**Comment by kirillrdy** [comment]
> Not sure whether hijacking the name is a good idea. Judging from a quick look at the new readme of wstunnel, the command line options are partially compatible (for instance -L remains the same), so scripts using wstunnel might break in unexpected ways a while after the update. IMO the best approach to this situation would be to first merge this PR, and then remove the old package and rewire the wstunnel module to the new version in a separate PR.

yes, sorry, hijacking is actually is a terrible idea from nixpkgs point of view, but from external point of view, both arch and void are somehow using wstunnel, either they never had haskell implementation or they reused the name

**Comment by alyaeanyx** [comment]
Yep, a quick look St the respective repositories tells me that neither had the original Haskell version in the first place

**Comment by SuperSandro2000** [comment]
wstunnel no longer builds on nixos-unstable, so we probably should hijack the name with a release not entry, remove the old package and update the module.

File: pkgs/by-name/ws/wstunnel-rust/package.nix [hunk 1]
```diff
@@ -0,0 +1,50 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "wstunnel-rust";
+  version = "9.2.2";
```
Comment (SuperSandro2000) [inline pkgs/by-name/ws/wstunnel-rust/package.nix:8]: Please update to https://github.com/erebe/wstunnel/releases/tag/v9.2.4

File: pkgs/by-name/ws/wstunnel-rust/package.nix [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      to constrain you to only use certain kind of protocols and consult
+      only a subset of the web. Nowadays, the most widespread protocol
+      is http and is de facto allowed by third party equipment.
+      Wstunnel uses the websocket protocol which is compatible with http
+      in order to bypass firewalls and proxies. Wstunnel allows you to
+      tunnel whatever traffic you want and access whatever resources/site
+      you need.
+      Note that this is the Rust rewrite of the original Haskell wstunnel.
+    '';
+    homepage = "https://github.com/erebe/wstunnel";
```
Comment (SuperSandro2000) [inline pkgs/by-name/ws/wstunnel-rust/package.nix:44]:
```suggestion
    homepage = "https://github.com/erebe/wstunnel";
    changelog = "https://github.com/erebe/wstunnel/releases/tag/${src.rev}";
```

File: pkgs/by-name/ws/wstunnel-rust/package.nix [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    license = licenses.bsd3;
+    mainProgram = "wstunnel";
+    maintainers = with maintainers; [ aleksana ];
+    platforms = platforms.unix;
```
Comment (SuperSandro2000) [inline pkgs/by-name/ws/wstunnel-rust/package.nix:48]:
```suggestion
```
that is the default for buildRustPackage

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
please adjust the module, too as mentioned in the comment above

**Pull request closed without merging**